cli/mysql.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/cheggaaa/pb"
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/docker/docker/pkg/term"
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-docopt"
+	"github.com/flynn/flynn/controller/client"
+	ct "github.com/flynn/flynn/controller/types"
+)
+
+func init() {
+	register("mysql", runMysql, `
+usage: flynn mysql console [--] [<argument>...]
+       flynn mysql dump [-q] [-f <file>]
+       flynn mysql restore [-q] [-f <file>]
+
+Options:
+	-f, --file=<file>  name of dump file
+	-q, --quiet        don't print progress
+
+Commands:
+	console  Open a console to a Flynn mysql database. Any valid arguments to mysql may be provided.
```

Comment:
@philiplb it would not block the way as any multi-engine support would be server side. The CLI is strictly client side.

That said, to address the compatibility side for completeness:

While the 2 aren't strictly compatible they are from a client perspective unless you are doing something very very strange. If you are worried about this you should read the "Incompatibilities between MariaDB 10.0 and MySQL 5.6"  section of the MariaDB/MySQL compatibility page: https://mariadb.com/kb/en/mariadb/mariadb-vs-mysql-compatibility/

In real terms there is no practical difference if you were already limiting yourself to the features of MySQL (note MariaDB has a much large feature set). Most of the incompatibilities relate to replication and other HA features which Flynn already takes care of so you shouldn't need to touch them.

We avoid most of the nasty stuff by not using Galera replication as it's write set replication while not strictly incompatible has very surprising behaviour. If you have heard of client issues with regards to MariaDB or PXC vs MySQL this is probably the source of that information. In Flynn we just use standard master/slave replication to avoid these very problems to ensure the clustering is as minimally impacting on client applications as possible.